crates/symbolicator/src/utils/sentry.rs

```diff
@@ -1,320 +1,79 @@
-    /// Returns the hub from a given http request.
-    ///
-    /// This requires that [`SentryMiddleware`] has been enabled or the call will panic.
-    fn from_request<S>(req: &HttpRequest<S>) -> Arc<Hub>;
-    /// Captures an actix error on the given hub.
-    fn capture_actix_error(&self, err: &Error) -> Uuid;
-}
-
-impl ActixWebHubExt for Hub {
-    fn from_request<S>(req: &HttpRequest<S>) -> Arc<Hub> {
-        req.extensions()
-            .get::<HubWrapper>()
-            .expect("SentryMiddleware middleware was not registered")
-            .hub
-            .clone()
-    }
-
-    fn capture_actix_error(&self, err: &Error) -> Uuid {
-        let mut exceptions = vec![];
-        let mut ptr: Option<&dyn Fail> = Some(err.as_fail());
-        let mut idx = 0;
-        while let Some(fail) = ptr {
-            // Check whether the failure::Fail held by err is a failure::Error wrapped in Compat
-            // If that's the case, we should be logging that error and its fail instead of the wrapper's construction in actix_web
-            // This wouldn't be necessary if failure::Compat<failure::Error>'s Fail::backtrace() impl was not "|| None",
-            // that is however impossible to do as of now because it conflicts with the generic implementation of Fail also provided in failure.
-            // Waiting for update that allows overlap, (https://github.com/rust-lang/rfcs/issues/1053), but chances are by then failure/std::error will be refactored anyway
-            let compat: Option<&failure::Compat<failure::Error>> = fail.downcast_ref();
-            let failure_err = compat.map(failure::Compat::get_ref);
-            let fail = failure_err.map_or(fail, |x| x.as_fail());
-            exceptions.push(exception_from_single_fail(
-                fail,
-                if idx == 0 {
-                    Some(failure_err.map_or_else(|| err.backtrace(), |err| err.backtrace()))
-                } else {
-                    fail.backtrace()
-                },
-            ));
-            ptr = fail.cause();
-            idx += 1;
-        }
-        exceptions.reverse();
-        self.capture_event(Event {
-            exception: exceptions.into(),
-            level: Level::Error,
-            ..Default::default()
-        })
-    }
-}
-
-fn exception_from_single_fail<F: Fail + ?Sized>(
-    f: &F,
-    bt: Option<&failure::Backtrace>,
-) -> Exception {
-    let dbg = format!("{:?}", f);
-    Exception {
-        ty: parse_type_from_debug(&dbg).to_owned(),
-        value: Some(f.to_string()),
-        stacktrace: bt
-            // format the stack trace with alternate debug to get addresses
-            .map(|bt| format!("{:#?}", bt))
-            .and_then(|x| parse_stacktrace(&x)),
+fn extract_request(req: &Request<Body>) -> (Option<String>, sentry::protocol::Request) {
+    let transaction = Some(req.uri().path().to_string());
+    let sentry_req = sentry::protocol::Request {
+        url: req.uri().to_string().parse().ok(),
```

Comment:
Does this extract e.g. request.scope nicely?

Reply:
Turned out `request.scope` came from somewhere else entirely :-D